ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParsePayload.java

```diff
@@ -204,14 +208,17 @@ private void addAttributesFromPayload(Map<String, String> attributes, ObjectNode
           .ifPresent(v -> attributes.put(Attribute.OS_VERSION, v));
     }
 
-    // Try extracting variants of top-level client id.
-    Stream
-        .of(attributes.get(Attribute.CLIENT_ID), json.path(Attribute.CLIENT_ID).textValue(),
-            json.path("clientId").textValue())
-        .map(ParsePayload::normalizeUuid) //
-        .filter(Objects::nonNull) //
-        .findFirst() //
-        .ifPresent(v -> attributes.put(Attribute.CLIENT_ID, v));
+    if (Objects.isNull(attributes.get(Attribute.CLIENT_ID))) {
+      // Try extracting variants of top-level client id and normalize UUID in JSON payload.
+      Stream.of(Attribute.CLIENT_ID, "clientId") //
```

Comment:
This works for now since the JSON structure is not nested. To update the json payload the field names are required. However, if in the future the structure gets more complex/nested then this needs to be changed.